File: ui/src/app/models/company.ts
import { Transform, Type } from 'class-transformer';
import { Account } from './account';
import { StatusTypes } from './enumTypes'
import { CompanyTypes } from './enumTypes'

export class Company {
	id?: number;
	name?: string;
	email?: string;
	phone?: string;
	label?: string;
	@Transform(({ value }) => { 
		if (StatusTypes[value as keyof typeof StatusTypes]) return StatusTypes[value as keyof typeof StatusTypes]; 
		else return null; 
	}, { toPlainOnly: true })
	status?: StatusTypes;
	@Transform(({ value }) => { 
		if (CompanyTypes[value as keyof typeof CompanyTypes]) return CompanyTypes[value as keyof typeof CompanyTypes]; 
		else return null; 
	}, { toPlainOnly: true })
	companyType?: CompanyTypes;
	@Type(() => Date)
	createdAt?: Date;
}

File: src/main/java/it/crm/domain/Company.java
package it.crm.domain;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import it.crm.enumerator.CompanyType;
import it.crm.enumerator.StatusType;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.OneToMany;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Company extends BaseEntity implements Serializable {

	private static final long serialVersionUID = -3702434523867111846L;
	
	private String name;
	private String email;
	private String phone;
	private String label;
	@Enumerated(EnumType.STRING)
	private StatusType status;
	@Enumerated(EnumType.STRING)
	private CompanyType companyType;
	@OneToMany(mappedBy = "company", cascade = CascadeType.ALL)
	private List<Responsability> responsabilities;
	
	@Override
	protected void softRemove(Date deleted) {
		
	}
}
